specs/backlog/node-owner-policy.rst

```diff
@@ -0,0 +1,195 @@
+..
+ This work is licensed under a Creative Commons Attribution 3.0 Unported
+ License.
+
+ http://creativecommons.org/licenses/by/3.0/legalcode
+
+======================================
+Adding Policy Targets for Ironic Nodes
+======================================
+
+Include the URL of your StoryBoard story (which should have an `rfe`` tag):
+
+https://storyboard.openstack.org/#!/story/XXXXXXX
+
+This spec describes an update to the Ironic node controller to expose Owner
+node data for policy checks.
+
+
+Problem description
+===================
+
+Right now Ironic is not multi-tenant; anyone with API access to one node
+has access to all nodes. Nodes have an ``owner`` field; however it is
+purely informational and not tied into any form of access control. This
+is inconvenient if an Ironic deployment consists of nodes from multiple
+hardware Owners: each Owner needs to be able to manage their own nodes
+through the Ironic API, while being unable to manage others' nodes.
+
+
+Proposed change
+===============
+
+The Ironic API already has a way of controlling access to the REST API: a
+policy engine[0] that is used throughout the REST API. However, when the node
+controller uses the policy engine[1], it does so without passing in any
+information about the node being checked. Other services such as Nova[2] and
+Barbican[3] pass in additional information, and we propose doing the same. In
+summary, we would like to:
```

Comment:
Nitpicks: we should fix the rst syntax here (and throughout) so that rather than footnotes we have active links.

Reply:
From what I've seen of other specs, they seem to prefer footnotes - maybe there's rst-specifc syntax for that I don't know about ... ?